nncf/quantization/algorithms/smooth_quant/openvino_backend.py

```diff
@@ -176,3 +146,16 @@ def get_weight_channel_axis(node: NNCFNode) -> int:
     @staticmethod
     def calculate_port_based_channel_axis(port_id: int, transpose: bool) -> int:
         return -2 + port_id if transpose else -1 - port_id
+
+    @staticmethod
+    def is_node_with_shared_weight(node: NNCFNode, nncf_graph: NNCFGraph) -> bool:
+        weight_port_id = OVSmoothQuantAlgoBackend.get_weight_tensor_port_id(node)
+        weight_node = nncf_graph.get_input_edges(node)[weight_port_id].from_node
+        return len(nncf_graph.get_next_nodes(weight_node)) > 1
+
+    @staticmethod
+    def get_filter_fn_for_statistics(activation_port_id: int) -> Callable[[StatisticPoint], bool]:
+        def filter_func(point: StatisticPoint) -> bool:
+            return point.target_point.port_id == activation_port_id
```

Comment:
Are you sure that we can safely remove the target point type check? Can't we mix up some statistics here?

Reply:
Yeah, you are right, I forgot to check target point type and algorithm key, thanks!
Fixed